gitdash/package.json

```diff
@@ -19,6 +19,7 @@
     "framer-motion": "^4.1.17",
     "next": "11.0.1",
     "next-auth": "^3.27.3",
+    "parse-link-header": "^1.0.1",
```

Comment:
Can you tell me more about this parse-link-header package?

Reply:
So basically there's no clear way to get the total number of starred repos. The way is you get it is what I mentioned in the PR description. Is making a simple request to that `starred` endpoint and in the headers you will get a field called `link`.
That header contains the true number of starred repos.

What that library does is it parses that header so you can get the actual total value which is `last_page` property as shown below:
`const starredCount = parsed.last.page;`	

Basically from the header you get this
`https://api.github.com/user/9287/repos?page=1&per_page=1; rel=head, https://api.github.com/user/9287/repos?page=88&per_page=1; rel=last`

What we need is this:
`page=88` which is the `rel=last` field which contains the actual total count.

That library parses the header to something like this (Going off my memory you can console.log if you wish):
`{ head:
   { page: '1',
     per_page: '1',
     rel: 'next',
     url: 'https://api.github.com/user/9287/repos?page=1&per_page=1' 
	},
  last:
   { page: '88',
     per_page: '1',
     rel: 'last',
     url: 'https://api.github.com/user/9287/repos?page=88&per_page=1' 
	} 
}`

In essence it's simply a parser and at first I spent like 2 hours looking for a simpler way we could parse and did find some suggestions but it was too much to implement because there are a lot of edge cases because the example is not the only type of header you can get. You can get no header if there are no starred repos and you can also get a header with an extra field. What that library does is handle all the edge-cases that we might miss. I also detest it because it doesn't have a typescript version. Maybe we can write our own parser when we find ourselves with some time.

I really don't like this solution as it's a bit hacky but the github api does have its limitations.

Reply:
Ah, so the number of pages is the number of starred repos? ngl, that's a pretty smart solution